### PR TITLE
CORE-14306. [NTOS:MM] Fix a Clang-Cl warning about ViewSize

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Assumed fix...

JIRA issue: [CORE-14306](https://jira.reactos.org/browse/CORE-14306)

## Proposed changes

- "warning: comparison 'ULONG_PTR' (aka 'unsigned long') > 4294967295 is always false [-Wtautological-constant-compare]"

## TODO

- [ ] Would you accept an additional commit to move 3 ">= 3" lines around to enforce the following pattern in `ARM3/pagfault.c`?
{{
#if (_MI_PAGING_LEVELS >= 3)
#if (_MI_PAGING_LEVELS == 4)
#endif
#endif
}}

### CODE REVIEW

**Comment by SergeGautherie** [comment]
For the record, there is a 4th warning, which I emailed @ThFabba about:
`ntoskrnl\mm\section.c(403,12) :  warning: variable 'ImageBase' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]`
See ROSTESTS-297 for some related documentation.

**Comment by ThFabba** [comment]
I tend to agree. While I'm not a fan of these extra parentheses, I think we have enough of them to warrant just disabling the warning.

File: ntoskrnl/mm/ARM3/section.c [hunk 1]
```diff
@@ -2896,7 +2896,10 @@ MmMapViewOfArm3Section(IN PVOID SectionObject,
         *ViewSize = (SIZE_T)(Section->SizeOfSection.QuadPart - SectionOffset->QuadPart);
 
         /* Check if it's larger than 4GB or overflows into kernel-mode */
-        if ((*ViewSize > 0xFFFFFFFF) ||
+        if (
+#ifdef _WIN64
+            (*ViewSize > 0xFFFFFFFF) ||
+#endif
```
[the remark this answers is not in the document]
Comment (ThFabba) [reply ntoskrnl/mm/ARM3/section.c]: This is obviously a no-op change so it doesn't need input from an x64 expert.

[the remark this answers is not in the document]
Comment (SergeGautherie) [reply ntoskrnl/mm/ARM3/section.c]: For what it's worth, line 1317 is `if (*ViewSize >= MAXLONG_PTR)`. I don't know whether the two lines should be related.

[the remark this answers is not in the document]
Comment (tkreuzer) [reply ntoskrnl/mm/ARM3/section.c]: This check is in fact totally wrong. It should check for an overflow on 32 bit architectures.
Should be:
```
    ULONG64 CalculatedViewSize;
    ....
    CalculatedViewSize = Section->SizeOfSection.QuadPart - SectionOffset->QuadPart;
    if (!NT_SUCCESS(RtlULongLongToSSIZET(CalculatedViewSize, ViewSize)) ||
        (....))
    {
        ...
    }
```
If that still causes a warning, we need to fix it in intsafe.h

**Comment by tkreuzer** [comment]
Replaced by PR #778

**Pull request closed without merging**